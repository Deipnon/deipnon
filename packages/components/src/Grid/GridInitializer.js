// @flow
import * as React from 'react'
import { Global, css } from '@emotion/core'
import theme from '@deipnon/themes/lib'

const globalStyles = css(theme.gridStyles)

const GridInitializer = () => <Global styles={globalStyles}/>

export default GridInitializer
