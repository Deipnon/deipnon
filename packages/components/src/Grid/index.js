// @flow
import React from 'react'
import styled from '@emotion/styled'
import { ClassNames } from '@emotion/core'

const ContainerElement = styled('div')``
const RowElement = styled('div')``
const ColumnElement = styled('div')``

type GridPropsType = {|
	children: React$Node,
	fluid: Boolean
|}

type RowPropsType = {|
	children: React$Node
|}

type ColPropsType = {|
	children: React$Node,
	lg?: Number,
	md?: Number,
	sm?: Number,
	xs?: Number

|}

class Grid extends React.PureComponent<GridPropsType> {
	static Row = (props: RowPropsType) => {
		return (<RowElement className="row">
			{props.children}
		</RowElement>)
	}

	static Col = (props: ColPropsType) => {
		const { xs, lg, md, sm, children } = props

		const className = {
			// $FlowFixMe
			[`col-xs-${xs}`]: !!xs,
			// $FlowFixMe
			[`col-sm-${sm}`]: !!sm,
			// $FlowFixMe
			[`col-md-${md}`]: !!md,
			// $FlowFixMe
			[`col-lg-${lg}`]: !!lg
		}

		return (<ClassNames>
			{ ({ css, cx }) => <ColumnElement className={cx(className)}>
				{children}
			</ColumnElement>}
		</ClassNames>)
	}

	container ({ fluid, children }: GridPropsType) {
		const className = {
			container: !fluid,
			'container-fluid': fluid
		}
		return (<ClassNames>
			{({ css, cx }) => {
				return (<ContainerElement className={cx(className)}>
					{children}
				</ContainerElement>)
			}}
		</ClassNames>)
	}

	render () {
		const Container = this.container

		return (<Container {...this.props}>
			{ this.props.children }
		</Container>)
	}
}

export default Grid
