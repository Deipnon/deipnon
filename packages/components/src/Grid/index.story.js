import React from 'react'
import { storiesOf } from '@storybook/react'
import GridInitializer from './GridInitializer'

import Grid from '.'

const stories = storiesOf('Grid', module)

const GridStoryComponent = () => (<React.Fragment>
	<GridInitializer />
	<Grid fluid>
		<Grid.Row>
			<Grid.Col xs={12} lg={6}>
				Deneme 1
			</Grid.Col>
			<Grid.Col xs={12} lg={6}>
				Deneme 2
			</Grid.Col>
		</Grid.Row>
	</Grid>
</React.Fragment>)

stories.add('Default', () => <GridStoryComponent />)
