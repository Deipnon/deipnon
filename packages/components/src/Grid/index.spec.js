import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-emotion'

import Grid from '.'
import GridInitializer from './GridInitializer'

describe('Grid', () => {
	test('renders correctly', () => {
		const tree = renderer.create(<React.Fragment>
			<GridInitializer />
			<Grid fluid>
				<Grid.Row>
					<Grid.Col xs={12} lg={6}>
				Deneme 1
					</Grid.Col>
					<Grid.Col xs={12} lg={6}>
				Deneme 2
					</Grid.Col>
				</Grid.Row>
			</Grid>
		</React.Fragment>).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
