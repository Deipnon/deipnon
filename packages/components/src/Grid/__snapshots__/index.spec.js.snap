// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<div
  className="container-fluid css-0"
>
  <div
    className="row css-0"
  >
    <div
      className="col-xs-12 col-lg-6 css-0"
    >
      Deneme 1
    </div>
    <div
      className="col-xs-12 col-lg-6 css-0"
    >
      Deneme 2
    </div>
  </div>
</div>
`;
